### DISCUSSION TITLE
Uncaught You must pass your app key when you instantiate Pusher.

### DISCUSSION THREAD

**Discussion by Shubhampandeysp01** [MusheAbdulHakim/Laravel-Smarthr#30] [General]
Getting this error on console: "Uncaught You must pass your app key when you instantiate Pusher." along with loading bar continuously just after login, looks like it got Stuck right after login. I have set the env variables for pusher after creating account. How to resolve this issue?
APP_KEY=base64:----------------------------=
REVERB_APP_ID=***********************
REVERB_APP_KEY=***********************
REVERB_APP_SECRET=***********************
REVERB_HOST="localhost"
REVERB_PORT=8080
REVERB_SCHEME=http

PUSHER_HOST="${REVERB_HOST}"
PUSHER_APP_KEY="${REVERB_APP_KEY}"
PUSHER_APP_SECRET="${REVERB_APP_SECRET}"
PUSHER_APP_ID="${REVERB_APP_ID}"
PUSHER_PORT="${REVERB_PORT}"
PUSHER_SCHEME="${REVERB_SCHEME}"

![image](https://github.com/user-attachments/assets/2e5a7b90-305a-4bf4-968e-37f6513bad0c)

**Comment by MusheAbdulHakim** [comment]
Please make sure reverb is installed and working correctly. If you are not going to use reverb, 
Set your pusher credentials correctly.

```
PUSHER_HOST=
PUSHER_APP_KEY=
PUSHER_APP_SECRET=
PUSHER_APP_ID=
PUSHER_PORT=
PUSHER_SCHEME=
```
And last make sure the APP_KEY is generated by running
```
php artisan key:generate
```
Checkout the reverb docs: https://laravel.com/docs/11.x/reverb#installation

**Comment by Shubhampandeysp01** [comment]
> Please make sure reverb is installed and working correctly. If you are not going to use reverb, Set your pusher credentials correctly.
> 
> ```
> PUSHER_HOST=
> PUSHER_APP_KEY=
> PUSHER_APP_SECRET=
> PUSHER_APP_ID=
> PUSHER_PORT=
> PUSHER_SCHEME=
> ```
> 
> And last make sure the APP_KEY is generated by running
> 
> ```
> php artisan key:generate
> ```
> 
> Checkout the reverb docs: https://laravel.com/docs/11.x/reverb#installation

Hey, I understood ur concern but i have installed reverb and pusher and have pusher credentials in my env file still i'm facing the issue. Can we connect for few mins to get this done please? I tried contacting you through ur telegram and whatsapp link provided in other issue but both ain't working.

**Comment by MusheAbdulHakim** [comment]
my whatsapp is active.

**Comment by MusheAbdulHakim** [comment]
if you want to connect with me, 
Via email: musheabdulhakim99@gmail.com
The Whatsapp number is: +233542441933.

**Comment by rehmatweb** [comment]
issue occurred after installing the reverb,

**Comment by MusheAbdulHakim** [comment]
I tried the installation process on a windows machine and didn't encounter this issue. I am inclined to believe that, this is an issue from the way you do the installations.
I made a video on how to do the installation of the application. Please check it out to find the solution to the problem. If you follow the instructions in the video, you won't have the issue.

Checkout the video here: [Youtube](https://youtu.be/UHkrsyBcMRM)
I am going to close this issue If are still having it, feel free to open the issue again or open a discussion for us to continue there. Thank you
